src/realm/object-store/collection.hpp

```diff
@@ -88,7 +88,20 @@ class Collection {
     // Return a Results representing a live view of this Collection.
     Results as_results() const;
 
-    NotificationToken add_notification_callback(CollectionChangeCallback cb) &;
+    /**
+     * Adds a `CollectionChangeCallback` to this `Collection`. The `CollectionChangeCallback` is exectuted when
+     * insertions, modifications or deletions happen on this `Collection`.
+     *
+     * @param callback The function to execute when a insertions, modification or deletion in this `Collection` was
+     * detected.
+     * @param key_path_array A filter that can be applied to make sure the `CollectionChangeCallback` is only executed
+     * when the property in the filter is changed but not otherwise.
+     *
+     * @return A `NotificationToken` that is used to identify this callback. This token can be used to remove the
+     * callback via `remove_callback`.
```

Comment:
`remove_callback()` is called by NotificationToken's destructor and isn't something that the caller of this can call (or know about).